R/wflow_toc.R

```diff
@@ -0,0 +1,45 @@
+#' Add a table of content on the clipboard
+#'
+#' \code{wfow_toc} add a table of content on the clipboard. The users can paste it
+#' in the \code{index.Rmd} document. The table of content includes all \code{html} document
+#' in the \code{docs} directory the users want to display.
+#'
+#' @param gh_url character. The Github Pages root directory.
+#' @param proj_url character. The project name.
+#'
+#' @return text character. The list of table of content in the markdown syntax.
+#' @import fs
+#' @import glue
+#' @import dplyr
+#' @import purrr
+#' @import readr
+#' @import tidyr
+#' @import clipr
+#' @import here
+#' @import rmarkdown
+#' @import tibble
+#' @export
+
+wflow_toc <- function(analysis_dir = "analysis", docs_dir = "docs") {
+  s <- wflow_status()
+  s$status %>%
+    tibble::rownames_to_column(var = 'rmd_path') %>%
+    dplyr::filter(published) %>%
+
+    dplyr::mutate(html_path = map_chr(rmd_path, to_html, outdir = s$docs),
+                  html_path = map_chr(html_path, fs::path_file)) %>%
+
+    dplyr::mutate(name = purrr::map(rmd_path,~ rmarkdown::yaml_front_matter(.)$title)) %>%
+    filter(!map_lgl(name, is_null)) %>%
+    tidyr::unnest() %>%
+
+    dplyr::mutate(
+      name = ifelse(is.na(name), html_path, name)
+    ) %>%
+    dplyr::mutate(
+      link = glue::glue("1. [{name}]({html_path})")
+    ) %>%
+    .$link %>%
+    clipr::write_clip()
```

Comment:
I'd like to make clipr a suggested package. If the user doesn't have clipr installed, then the function can print the results to the R console. Either way it would be good to return the character vector invisibly using `return(invisible(x))`, where `x` would be replaced with the character vector containing the toc.

You can check if a package is installed using the following strategy:

```
if requireNamespace("clipr", quietly = TRUE) {
  # Write to clipboard
} else {
  # Print to R console
}
```

Reply:
I use your recommendation, but I do it in my way because I still get the point of the function `invisible`.
In my understanding, if the users have already installed the package `clipr`, they will give the toc in their clipboard. If not, the function `wflow_toc` will print the toc in the console?

Do I miss something?